pkg/gofr/gofr.go

```diff
@@ -177,6 +178,29 @@ func (a *App) Run() {
 	wg.Wait()
 }
 
+// Shutdown stops the service(s) and close the application.
+func (a *App) Shutdown(ctx context.Context) (err error) {
```

Comment:
It seems to me you didn't implement a timeout processing as discussed in issue

https://github.com/gofr-dev/gofr/issues/663#issuecomment-2135108278

Maybe you would in another PR ?